crates/wasm/fs/private/forest.rs

```diff
@@ -23,4 +35,48 @@ impl PrivateForest {
     pub fn new() -> PrivateForest {
         Self(Rc::new(WnfsPrivateForest::default()))
     }
+
+    #[wasm_bindgen]
+    pub fn get(
+        &self,
+        saturated_namefilter_hash: Uint8Array,
+        revision_key: Uint8Array,
+        store: BlockStore,
+    ) -> JsResult<Promise> {
+        let store = ForeignBlockStore(store);
+        let forest = self.0.clone();
+
+        let saturated_name_hash = expect_bytes::<HASH_BYTE_SIZE>(saturated_namefilter_hash)?;
+
+        let key_bytes = expect_bytes::<KEY_BYTE_SIZE>(revision_key)?;
+        let key = Key::new(key_bytes);
+        let ratchet_key = RatchetKey(key);
+
+        let private_ref = PrivateRef::from_ratchet_key(saturated_name_hash, ratchet_key);
+
+        Ok(future_to_promise(async move {
+            let node_option = forest
+                .get(&private_ref, &store)
+                .await
+                .map_err(error("Cannot 'get' in forest"))?;
+
+            Ok(match node_option {
+                Some(node) => value!(PrivateNode(node)),
+                None => JsValue::NULL,
+            })
+        }))
+    }
+}
+
+fn expect_bytes<'a, const N: usize>(bytes: Uint8Array) -> JsResult<[u8; N]> {
+    let vec = bytes.to_vec();
+    if vec.len() != N {
+        return Err(Error::new(&format!(
+            "Unexpected number of bytes received. Expected {N}, but got {}",
+            vec.len()
+        )));
+    }
+    let mut slice = [0u8; N];
+    slice.copy_from_slice(vec.as_slice());
+    Ok(slice)
 }
```

Comment:
This can then become:

```suggestion
#[inline]
fn expect_bytes<const N: usize>(bytes: Vec<u8>) -> JsResult<[u8; N]> {
    bytes.try_into().map_err(|v: Vec<u8>| {
        Error::new(&format!(
            "Unexpected number of bytes received. Expected {N}, but got {}",
            v.len()
        ))
    })
}
```

Reply:
Thanks, that's much better.